packages/grid/_modules_/grid/models/api/i18nApi.ts

```diff
@@ -0,0 +1,13 @@
+export type LocaleTextValue = string | React.ReactNode | Function;
+
+/**
+ * The i18n API interface that is available in the grid [[apiRef]].
+ */
+export interface I18nApi {
+  /**
+   * Get grid text.
+   * @param key
+   * @returns LocaleTextValue
+   */
+  getText: (key: string) => LocaleTextValue;
```

Comment:
I think that this type is wrong, it should retain the type of the key in LocaleText.

Reply:
This only defines what can be returned. I can remove the return type and put it as any but other than that I don't think I can make it dependent on the type of the object key. I don't have access to `key` when determining the return type

Reply:
I have pushed a change that restricts the types to be closer to the underlying implementation.